src/index.js

```diff
@@ -1098,42 +1098,36 @@ export function addAction (name, opts) {
     opts.response = opts.response || function (response) { return response }
     opts.responseError = opts.responseError || function (err) { return utils.reject(err) }
     mapper[name] = function (id, _opts) {
-      if (utils.isObject(id)) {
-        _opts = id
-      }
```

Comment:
I think we still need these 3 lines. To use your test as an example, `getTeacherReportsAsync(1234, { params: { foo: 'bar' } })` should produce `async/reports/schools/1234/teachers?foo=bar` like you've noted, but `getTeacherReportsAsync({ params: { foo: 'bar' } })` should produce `async/reports/schools/teachers?foo=bar`. Without these 3 lines the second example would not work.